crates/ntx-builder/src/builder/mod.rs

```diff
@@ -0,0 +1,96 @@
+use std::{net::SocketAddr, time::Duration};
+
+use anyhow::Context;
+use miden_node_proto::generated::ntx_builder::api_server;
+use server::NtxBuilderApi;
+use store::StoreClient;
+use tokio::{net::TcpListener, time};
+use tokio_stream::wrappers::TcpListenerStream;
+use tower_http::trace::TraceLayer;
+use tracing::{debug, info, warn};
+
+use crate::COMPONENT;
+
+mod server;
+
+/// Interval for checking pending notes
+const NOTE_CHECK_INTERVAL_MS: u64 = 100;
+
+//mod data_store;
+mod store;
+
+/// Network Transaction Builder
+pub struct NetworkTransactionBuilder {
+    /// The address of the network transaction builder gRPC server.
+    pub address: SocketAddr,
+    /// Address of the store gRPC server.
+    pub store_address: SocketAddr,
+}
+
+impl NetworkTransactionBuilder {
+    /// Serves the network transaction builder RPC API
+    ///
+    /// Note: Executes in place (i.e. not spawned) and will run indefinitely until
+    ///       a fatal error is encountered.
+    pub async fn serve(self) -> anyhow::Result<()> {
+        info!(
+            target: COMPONENT,
+            endpoint=?self.address,
+            "Starting network transaction builder server"
+        );
+
+        let store = StoreClient::new(self.store_address);
+        let unconsumed_network_notes = store.get_unconsumed_network_notes().await?;
+
+        let api = NtxBuilderApi::new(unconsumed_network_notes);
+        let api_state = api.state();
+        let api_service = api_server::ApiServer::new(api);
+
+        let listener = TcpListener::bind(self.address).await?;
+
+        // Spawn the ticker task to process pending notes periodically
+        let tick_handle = tokio::spawn(async move {
+            let mut interval = time::interval(Duration::from_millis(NOTE_CHECK_INTERVAL_MS));
+
+            loop {
+                interval.tick().await;
+
+                {
+                    let state_guard = match api_state.lock() {
+                        Ok(guard) => guard,
+                        Err(e) => {
+                            warn!(target: COMPONENT, error=%e, "Failed to lock API state - poisoned lock");
+                            continue;
```

Comment:
I think I have seen us `expect()` on locks like this in the code base. I think its OK to panic in that instance - but it depends on how important graceful shutdown (or the absence of it) is.